crates/provider_openai/src/speech.rs

```diff
@@ -0,0 +1,40 @@
+use axum::body::Bytes;
+use reqwest::Client;
+use serde_json::json;
+use unspeech_shared::{
+  AppError,
+  speech::ProcessedSpeechOptions,
+};
+
+pub async fn handle(
+  options: ProcessedSpeechOptions,
+  client: Client,
+  token: &str,
+) -> Result<Bytes, AppError> {
+  let body = json!({
+    "input": options.input,
+    "model": options.model,
+    "voice": options.voice,
+    "instructions": options.instructions,
+    "response_format": options.response_format,
+    "speed": options.speed,
+  });
+
+  let res = client.post("https://api.openai.com/v1/audio/speech")
+    .bearer_auth(token)
+    .json(&body)
+    .send()
+    .await?;
+
+  if !res.status().is_success() {
+    let status = res.status();
+    let body = res.text().await.unwrap_or_else(|_| "Could not read error body".to_string());
```

Comment:
Do we want to print the detailed error here?

```suggestion
    let body = res.text().await.unwrap_or_else(|err| format!("Could not read error body: {}", err));
```